File: C/pub1/Distrib/LIB/LIB-PHP/Example/subp.ping.run.php
#!php.exe
<?php
/* ********************************************************
 * subp.ping.run.php
 * This Script will Run a ping.exe Tool
**********************************************************/

# Setup Command Interpreter
$pathCMD = $_SERVER['SystemRoot'] . "\\System32";
$CMDEXE = $pathCMD . "\\" . "cmd.exe";
#$pingHost = "localhost";
#$aPingCmd = "ping.exe -n 4 -6 ";
$pingHost = "google.com";
$aPingCmd = "ping.exe -n 4 -4 ";
if( is_dir($pathCMD)) {
    if(is_file($CMDEXE)){
        $strValue = $aPingCmd . $pingHost;
        $handle = (popen($strValue, 'r'));
        echo( "\n" . '$handle:' . gettype($handle) . "\n");
        while(!feof($handle)){
            $read = fread($handle, 4096);
            echo($read);
        }
        $r = pclose($handle);
        $rjson = json_encode(array('result'=>$r));
        print_r($rjson);
    }
    else {
        echo("\nA File " . $CMDEXE . " is not found.");
    }
}
else {
    echo("\nA Path " . $pathCMD . " is not Found.");
}
?>

File: C/pub1/Distrib/LIB/LIB-RUBY/Example/subp.ping.run.rb
#!ruby.exe
#* ********************************************************
#* subp.ping.run.php
#* This Script will Run a ping.exe Tool
#*********************************************************/

require 'open3'

# Setup Command Interpreter
pathCMD = ENV['SystemRoot'] + "\\System32"
cmdEXE = pathCMD + "\\" + "cmd.exe"
# pingHost = "localhost"
# aPingCmd = " -n 4 -6 "
pingHost = "google.com"
aPingCmd = " -n 4 -4 "
aPingExe = pathCMD + "\\ping.exe"
if File.directory?(pathCMD) then
    if File.file?(cmdEXE) then
        strValue = aPingCmd + pingHost
        # puts aPingExe
        if File.file?(aPingExe) then
            stdout,stderr,status=Open3.capture3(aPingExe + strValue)
            puts "\nSTDOUT=\n"
            puts stdout
            puts "\nSTDERR=\n"
            puts stderr
            puts "\nResult: #{status.exitstatus}"
        else
            puts "\nA File #aPingExe is not Found."
        end
    else
        puts "\nA File #cmdExe is not Found."
    end
else
    puts "\nA Folder #pathCMD is not Found."
end